openshift/templates/api-deployment-config.yaml

```diff
@@ -0,0 +1,168 @@
+kind: Template
+apiVersion: template.openshift.io/v1
+metadata:
+  name: ${CONTAINER_NAME}-template
+labels:
+  template: ${CONTAINER_NAME}-template
+
+objects:
+  - kind: DeploymentConfig
+    apiVersion: apps.openshift.io/v1
+    metadata:
+      annotations:
+        app.openshift.io/route-disabled: 'false'
+        openshift.io/generated-by: OpenShiftWebConsole
+      name: ${CONTAINER_NAME}
+      namespace: ${OPENSHIFT_PROJECT}
+      labels:
+        app: ${CONTAINER_NAME}
+        app.kubernetes.io/component: ${CONTAINER_NAME}
+        app.kubernetes.io/instance: ${CONTAINER_NAME}
+        app.kubernetes.io/name: ${CONTAINER_NAME}
+        app.kubernetes.io/part-of: ${APPLICATION_NAME}
+        app.openshift.io/runtime: nodejs
+        app.openshift.io/runtime-namespace: ${OPENSHIFT_PROJECT}
+    spec:
+      strategy:
+        type: Rolling
+        rollingParams:
+          updatePeriodSeconds: 1
+          intervalSeconds: 1
+          timeoutSeconds: 600
+          maxUnavailable: 25%
+          maxSurge: 25%
+        resources: {}
+        activeDeadlineSeconds: 21600
+      triggers:
+        - type: ImageChange
+          imageChangeParams:
+            automatic: true
+            containerNames:
+              - ${CONTAINER_NAME}
+            from:
+              kind: ImageStreamTag
+              namespace: ${OPENSHIFT_PROJECT}
+              name: '${CONTAINER_NAME}:latest'
+            lastTriggeredImage: >-
+              ${IMAGE_REPOSITORY}/${OPENSHIFT_PROJECT}/${CONTAINER_NAME}
```

Comment:
```suggestion
```

Reply:
you won't need this if you're using artifactory

Reply:
no the other item is config changes only. so if you manually changed an env var on the dc directly in OC, it would redeploy.